### ISSUE TITLE
OOM due to too big batch_size

### ISSUE THREAD

**Issue by RomainGoussault** [SubstraFoundation/distributed-learning-contributivity#137]
I had the following OOM (out of memory) error :
```
tensorflow.python.framework.errors_impl.ResourceExhaustedError:  OOM when allocating tensor with shape[3600,64,24,24] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
	 [[node gradients/max_pooling2d_1/MaxPool_grad/MaxPoolGrad (defined at /data/install/anaconda/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py:1751) ]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.
 [Op:__inference_keras_scratch_graph_11776773]

```

The batch_size (3600) was too big. One parter had 60% of the data, resulting in 28 000 training samples. With a gradient_fit_count of 8, it results a batch size of 28 000 / 8 = 3600.

**Comment by RomainGoussault** [comment]
Should we impose a hard upper limit on the batch size ? @bowni

**Comment by bowni** [comment]
I think that is a good idea. Or we can document in the config files that the user should have in mind that the combination of the mini-batch count and the gradient updates per pass count has to be carefully thought of. Or both!

**Comment by RomainGoussault** [comment]
Fixed in #149
